### PR TITLE
Add "Google Search" Extension

### PR DESCRIPTION
**Pull request by mblode**
## Description

This is a new extensions that adds "Google search" extensions

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/7183998/148449524-2558f2dc-9ae9-4f90-a381-ffebc175a1e8.mp4

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Comment by peteschaffner** [comment]
@mblode Thank you so much for this extension!

I just wanted to mention that it will fail, however, if you have the "Limit IP Address Tracking" setting enabled in System Preferences > Network. It might be worth mentioning in a README or something.

**Comment by mblode** [comment]
@peteschaffner thanks for that. I've updated the Extension and added that as a note in the readme.

I've also add search history as a listing when the user hasn't typed anything which is handled via local storage. Shout out to Ayush for the suggestion. The history can be cleared and individual results can be removed

File: extensions/google-search/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "title": "Google Search",
+  "description": "Google search with autosuggestions",
+  "icon": "command-icon.png",
+  "author": "mblode",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Google or type a URL",
+      "subtitle": "Google",
```
Comment (pernielsentikaer) [inline extensions/google-search/package.json:13]:
```suggestion
      "title": "Google Search",
```
Normally we use the subtitle, but in this case it doesn't make sense

File: extensions/google-search/src/utils/handleResults.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    return [];
+  }
+
+  const items: SearchResult[] = JSON.parse(historyString);
+  return items;
+}
+
+export async function getSearchResults(searchText: string, signal: AbortSignal): Promise<SearchResult[]> {
+  const response = await fetch(
+    `http://suggestqueries.google.com/complete/search?hl=en-us&output=chrome&q=${encodeURIComponent(searchText)}`,
```
Comment (pernielsentikaer) [inline extensions/google-search/src/utils/handleResults.ts:18]:
```suggestion
    `https://suggestqueries.google.com/complete/search?hl=en-us&output=chrome&q=${encodeURIComponent(searchText)}`,
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @mblode 👋 

Thanks for your contribution, exciting extension! 🔥 

I left a few suggestions and have a single thing I noticed, It seems like there is something with the encoding in some occasions

xvO6BV9lS0BfsCOL

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by peteschaffner** [comment]
> @peteschaffner thanks for that. I've updated the Extension and added that as a note in the readme. Also I'm a massive fan of Tower! Best Git GUI and I've been using it for years ❤️

Great to hear Tower has been serving you well for so many years!!

BTW your extension has inspired me to [eventually] create a similar one for DuckDuckGo :)

**Comment by mblode** [comment]
Note: I'll split this out into 2 pull requests tomorrow morning and respond to the feedback

File: extensions/google-search/src/utils/handleResults.ts (hunk 2 above)
Comment (mblode) [reply extensions/google-search/src/utils/handleResults.ts]: Fixed 👍

File: extensions/google-search/package.json (hunk 1 above)
Comment (mblode) [reply extensions/google-search/package.json]: Fixed 👍

**Comment by mblode** [comment]
@pernielsentikaer I've resolved the issues you've raised:

- Fixed the text encoding issues by using Latin-1 (`iso-8859-1`) instead of UTF-8 for the results. E.g. "æ" will encode correctly 
- Set it to fetch from `https` instead of `http`.
- Updated the command title to "Google Search" and removed the subtitle

**Review by pernielsentikaer — APPROVED** [review]
Hi @mblode 👋 

MUCH better, I can't find anything else that I would like to have changed before accepting it. 

I'm going to accept it and then will my colleague do a final review before it's finally merged.

**Comment by lukerooneyio** [comment]
@mblode Hi Matt, would it be possible to have the extension sort links at the top? I'd find it more useful such as the cases I've shown below. Besides that, thank you for this extension. This is what I wanted from Alfred years ago!!


Screen Shot 2022-01-13 at 10 47 55 PM
Screen Shot 2022-01-13 at 10 48 04 PM
e

**Comment by KnifeFed** [comment]
Would it be possible to implement something akin to tab completion? I.e. the ability to choose "raycast google search" here and then continue typing to add e.g. "extension":

CleanShot 2022-01-30 at 01 26 40@2x

**Pull request merged**